src/Appwrite/Platform/Workers/Functions.php

```diff
@@ -130,6 +130,11 @@ public function action(Message $message, Database $dbForProject, Func $queueForF
                     if (!array_intersect($events, $function->getAttribute('events', []))) {
                         continue;
                     }
+                    // filter disabled functions from getting executed.
+                    if (!$function->getAttribute('enabled')) {
```

Comment:
```suggestion
                    if ($function->getAttribute('enabled', true) === false) {
```
this would be a more strict check.